itest/rust/src/runner.rs

```diff
@@ -66,8 +69,18 @@ impl IntegrationTests {
         self.conclude(rust_time, gdscript_time, allow_focus)
     }
 
-    fn run_rust_tests(&mut self, tests: Vec<RustTestCase>, scene_tree: Gd<Node>) {
-        let ctx = TestContext { scene_tree };
+    fn run_rust_tests(
+        &mut self,
+        tests: Vec<RustTestCase>,
+        root_node: Gd<Window>,
+        test_node: Gd<Node>,
+        scene_tree: Gd<SceneTree>,
+    ) {
+        let ctx = TestContext {
+            _scene_tree: scene_tree,
+            root_node,
+            test_node,
+        };
```

Comment:
I don't think the new fields in `TestContext` are necessary:
* `_scene_tree` is unused
   * the user can always call `Node::get_tree()` if absolutely necessary
* `root_node` is only used for `push_input()`
   * Must this happen on the _root_ node? Can it not be the node itself or its parent? I don't want to generally encourage people to touch the scene tree too much, as we have zero test isolation and any side effects are persisted. Of course we cannot _prevent_ it, but we should also not serve things on a silver platter 😉 
   * If it's needed, we can also consider `get_tree()` + `get_root()`, even if it's a bit ugly with intermediate unwraps.

Also, please don't rename `scene_tree` -> `test_node` while there are so many PRs open, this creates extra conflicts for little benefit. I can take care of this later if necessary. The main idea was that you would get access to the scene tree through that field, even if it's technically not a `SceneTree` but a `Node`. Maybe `test_node` is clearer though.

Reply:
> I don't think the new fields in `TestContext` are necessary:
> 
>     * `_scene_tree` is unused

fair, i kept it cause it was the old name of the field, which was confusing cause it wasn't actually the scene tree.

>     * `root_node` is only used for `push_input()`
>       
>       * Must this happen on the _root_ node? Can it not be the node itself or its parent? I don't want to generally encourage people to touch the scene tree too much, as we have zero test isolation and any side effects are persisted. Of course we cannot _prevent_ it, but we should also not serve things on a silver platter 😉
>       * If it's needed, we can also consider `get_tree()` + `get_root()`, even if it's a bit ugly with intermediate unwraps.
> 

`push_input` must be done on a viewport. i guess it might be possible to make a viewport and add the node as a child and call push_input? ill check if that's possible.

> Also, please don't rename `scene_tree` -> `test_node` while there are so many PRs open, this creates extra conflicts for little benefit. I can take care of this later if necessary. The main idea was that you would get access to the scene tree through that field, even if it's technically not a `SceneTree` but a `Node`. Maybe `test_node` is clearer though.

fair, it was very confusing to me because i thought it was the scene tree at first but it actually isnt...